Source/effects.cpp

```diff
@@ -896,182 +896,111 @@ bool __fastcall effect_is_playing(int nSFX)
+				Monsters[monst].Snds[i][j] = pSnd;
+				if ( !pSnd )
+					mem_free_dbg(path);
 			}
-			++v1;
 		}
-		while ( v1 < 4 );
 	}
 }
 
 void __cdecl FreeEffects()
 {
-	TSnd **v0; // esi
-	signed int v1; // ebp
-	signed int v2; // ebx
-	TSnd *v3; // ecx
-	void *v4; // edi
-	TSnd **v5; // [esp+0h] [ebp-8h]
-	int v6; // [esp+4h] [ebp-4h]
-
-	v6 = 0;
-	if ( nummtypes > 0 )
-	{
-		v5 = Monsters[0].Snds;
-		do
-		{
-			v0 = v5;
-			v1 = 4;
-			do
-			{
-				v2 = 2;
-				do
-				{
-					v3 = *v0;
-					if ( *v0 )
-					{
-						*v0 = 0;
-						v4 = (void *)v3->sound_path;
-						v3->sound_path = 0;
-						sound_file_cleanup(v3);
-						mem_free_dbg(v4);
-					}
-					++v0;
-					--v2;
+	for ( int i = 0; i < nummtypes; i++ ) {
+		int mtype = Monsters[i].mtype;
+		for ( int j = 0; j < 4; ++j ) {
+			for ( int k = 0; k < 2; ++k ) {
+				TSnd *pSnd = Monsters[i].Snds[j][k];
+				if ( pSnd ) {
+					Monsters[i].Snds[j][k] = 0;
+					char *file = pSnd->sound_path;
+					pSnd->sound_path = 0;
+					sound_file_cleanup(pSnd);
+					mem_free_dbg(file);
 				}
-				while ( v2 );
-				--v1;
 			}
-			while ( v1 );
-			++v6;
-			v5 += 82;
 		}
-		while ( v6 < nummtypes );
 	}
 }
 
 void __fastcall PlayEffect(int i, int mode)
 {
-	int v2; // edi
-	int v3; // esi
-	int v4; // eax
-	int v5; // esi
-	int v6; // eax
-	TSnd *v7; // edi
-	//int v8; // eax
-	int volume_delta; // [esp+8h] [ebp-8h]
-	int pan; // [esp+Ch] [ebp-4h]
+	if ( plr[myplr].pLvlLoad ) {
+		return;
+	}
 
-	v2 = mode;
-	v3 = i;
-	if ( !plr[myplr].pLvlLoad )
-	{
-		v4 = random(164, 2);
-		if ( gbSndInited )
-		{
-			if ( gbSoundOn )
-			{
-				if ( !gbBufferMsgs )
-				{
-					v5 = v3;
-					v6 = v4 + 2 * (v2 + 41 * monster[v5]._mMTidx);
-					v7 = Monsters[0].Snds[v6];
-					if ( v7 )
-					{
-						//_LOBYTE(v8) = snd_playing(Monsters[0].Snds[v6]);
-						if ( !snd_playing(Monsters[0].Snds[v6]) )
-						{
-							if ( calc_snd_position(monster[v5]._mx, monster[v5]._my, &volume_delta, &pan) )
-								snd_play_snd(v7, volume_delta, pan);
-						}
-					}
-				}
-			}
-		}
+	int sndIdx = random(164, 2);
+	if ( !gbSndInited || !gbSoundOn || gbBufferMsgs ) {
+		return;
+	}
+
+	int mi = monster[i]._mMTidx;
+	TSnd *snd = Monsters[mi].Snds[mode][sndIdx];
+	if ( !snd || snd_playing(snd) ) {
+		return;
 	}
+
+	int lVolume, lPan;
+	if ( !calc_snd_position(monster[i]._mx, monster[i]._my, &lVolume, &lPan) )
+		return;
+
+	snd_play_snd(snd, lVolume, lPan);
 }
 // 4A22D5: using guessed type char gbSoundOn;
 // 676194: using guessed type char gbBufferMsgs;
 
-int __fastcall calc_snd_position(int x, int y, int *plVolume, int *plPan)
+BOOL __fastcall calc_snd_position(int x, int y, int *plVolume, int *plPan)
 {
-	int v4; // edi
-	int v5; // esi
-	int v6; // eax
-	int v7; // ebx
-	int v8; // eax
-	int v9; // eax
+	x -= plr[myplr].WorldX;
+	y -= plr[myplr].WorldY;
+
+	int pan = (x - y) << 8;
+	*plPan = pan;
+
+	if ( abs(pan) > 6400 )
+		return FALSE;
+
+	int volume = abs(x) > abs(y) ? abs(x) : abs(y);

```

Comment:
This was probably a `max` macro.

```c
int volume = max(abs(x), abs(y);
```

Reply:
Thanks, I didn't realize that C had max() wound have been helpful in other situations, maybe I will revisit player.cpp now :)